src/control/server/instance.go

```diff
@@ -274,6 +274,11 @@ func (ei *EngineInstance) handleReady(ctx context.Context, ready *srvpb.NotifyRe
 }
 
 func (ei *EngineInstance) SetupRank(ctx context.Context, rank ranklist.Rank, map_version uint32) error {
+	if ei.IsReady() {
+		ei.log.Errorf("SetupRank called on an already set-up instance %d", ei.Index())
+		return nil
+	}
+
```

Comment:
Does this PR address the race condition, or do we just log an error on duplicate calls?

Reply:
Should it be an error or warning? I'm just thinking from the perspective of an Admin wondering why they see this error if it's not _**really**_ an issue

Reply:
Sure, not an issue from me

Reply:
I've updated the log level for this entry to Debug in this PR